### DISCUSSION TITLE
Json from mmap: get the pointer of the data

### DISCUSSION THREAD

**Discussion by ghost** [nlohmann/json#3515] [Q&A] [answered]
Hi,
I load my json from a mmap file.
I need to report back the modifications I do to that file.

What is the best way?

An alternative would be to get the binary pointer of that value in the mmap file.
The problem is that I need to copy a string representing a float from my app in a textual value of the json object.

Thanks,

**Comment by ghost** [comment]
Basically, in-place update is what I am after.

**Comment by falbrechtskirchinger** [comment, marked as answer]
When a given input is parsed, no metadata is stored in the object to link it back to its input. To some limited extent, you might be able to track modifications by providing your own object and array types. Modifications of primitive types would have to be tracked through their parent object or array which isn't going to be reliable.
Beyond that, the library has no mechanism to hook into the serialization to make use of that modification data.

I'd say what you're after isn't possible without significant work.

**Reply by ghost** [reply]
Thanks for your input.
Then I will store the strings in an unordered_map, inject a known value in the json object which will be my key, dump it, process the string with a regex and manually replace the occurrence with raw pointers in my string from my unordered_map.
Et voilà. It is offline work so efficiency comes second here. 🙂
